### PR TITLE
Adding MuArmGicExLib and update the usage accordingly

### PR DESCRIPTION
**Pull request by kuqin12**
## Description

This change introduced the MuArmGicExLib into MsCorePkg to host a few interfaces and definitions EDK2 prefer not to support due to lack of active consumers.

The consumers in Project MU (MpManagement driver and app) are updated to use the new interface and library implementation.

- [x] Impacts functionality?
- [ ] Impacts security?
- [x] Breaking change?
- [ ] Includes tests?
- [ ] Includes documentation?

## How This Was Tested

Tested on pipeline server build.

## Integration Instructions

Platforms using MpManagement functionality should include `MuArmGicExLib` in their DSC file.

### CODE REVIEW

File: MsCorePkg/Include/Library/MuArmGicExLib.h
Comment (makubacki) [inline MsCorePkg/Include/Library/MuArmGicExLib.h:1]: Can you elaborate on why edk2 did not accept the change in the PR description? That's good to know of course, but the next question is why.

File: MsCorePkg/Library/MuArmGicExLib/AArch64/MuArmGicEx.masm
Comment (makubacki) [inline MsCorePkg/Library/MuArmGicExLib/AArch64/MuArmGicEx.masm:1]: Can you please update they copyright header to use the newer format with the SPDX identifier?

File: MsCorePkg/Library/MuArmGicExLib/MuArmGicExLib.c [hunk 1]
```diff
@@ -0,0 +1,299 @@
+/**
+  @file  MuArmGicExLib.c
+
+  This library implements the extended interfaces for the Arm GIC.
+
```
Comment (makubacki) [inline MsCorePkg/Library/MuArmGicExLib/MuArmGicExLib.c:5]: Missing copyrights and license.

File: MsCorePkg/Library/MuArmGicExLib/MuArmGicExLib.c [hunk 2]
```diff
@@ -0,0 +1,299 @@
...
+          ARM_GICR_CTLR_FRAME_SIZE + ARM_GICR_ICPENDR + 4 * (offset))
+
+/**
+ * Return the base address of the GIC redistributor for the current CPU
+ *
+ * @param Revision  GIC Revision. The GIC redistributor might have a different
+ *                  granularity following the GIC revision.
+ *
+ * @retval Base address of the associated GIC Redistributor
+ */
```
Comment (makubacki) [inline MsCorePkg/Library/MuArmGicExLib/MuArmGicExLib.c]:
```suggestion
/**
  Return the base address of the GIC redistributor for the current CPU

  @param[in] GicRedistributorBase The base address of the GIC Distributor.
  @param[in] Revision             GIC Revision. The GIC redistributor might have a different
                                  granularity following the GIC revision.

  @retval Base address of the associated GIC Redistributor
**/
```

**Review by makubacki — APPROVED** [review]

File: UefiTestingPkg/FunctionalSystemTests/MpManagement/Driver/AARCH64/SuspendHandling.c [hunk 3]
```diff
@@ -359,7 +360,7 @@ CpuArchWakeFromSleep (
 {
   // Sending SGI to the specified secondary CPU interfaces
   // TODO: This is essentially reverse engineering to correlate the CPU index with the MPIDRs...
```
Comment (apop5) [inline UefiTestingPkg/FunctionalSystemTests/MpManagement/Driver/AARCH64/SuspendHandling.c]: Is this TODO still needed?

**Review by apop5 — APPROVED** [review]

File: MsCorePkg/Include/Library/MuArmGicExLib.h
Comment (kuqin12) [reply MsCorePkg/Include/Library/MuArmGicExLib.h:1]: There is no active consumer of these interfaces and definitions in the codebase is mainly why. The only send SGI function that is used by MP core on ARM platform will be removed by Ard according to the conversation in EDK2 PR: https://github.com/tianocore/edk2/pull/5739.

File: MsCorePkg/Library/MuArmGicExLib/MuArmGicExLib.c (hunk 1 above)
Comment (kuqin12) [reply MsCorePkg/Library/MuArmGicExLib/MuArmGicExLib.c:5]: Thanks, updated!

File: UefiTestingPkg/FunctionalSystemTests/MpManagement/Driver/AARCH64/SuspendHandling.c (hunk 3 above)
Comment (kuqin12) [reply UefiTestingPkg/FunctionalSystemTests/MpManagement/Driver/AARCH64/SuspendHandling.c]: Nope, removed.

File: MsCorePkg/Library/MuArmGicExLib/AArch64/MuArmGicEx.masm
Comment (kuqin12) [reply MsCorePkg/Library/MuArmGicExLib/AArch64/MuArmGicEx.masm:1]: Updated!

**Pull request merged**